### PR TITLE
Refactor dropdown body and add buttons customization

### PR DESCRIPTION
**Pull request by ivanmjartan**
JIRA: RAIL-4324

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8571/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10422/ : SUCCESS in 12m 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8572/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10423/ : SUCCESS in 12m 26s

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/AttributeFilterDropdownBody.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// (C) 2022 GoodData Corporation
+import React, { useCallback } from "react";
+import {
+    AttributeFilterDropdownContent,
+    IAttributeDropdownBodyPropsNoCallbacks,
+} from "./AttributeFilterDropdownContent";
+import { useAttributeFilterComponentsContext } from "../Context/AttributeFilterComponentsContext";
+
+interface IAttributeFilterDropdownBodyProps {
+    allElementsFiltered: boolean;
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/AttributeFilterDropdownBody.tsx:10]: maybe we should align it with `hasNoData` ? (eg `hasNoMatchingData`)

File: libs/sdk-ui-filters/src/AttributeFilter@next/AttributeFilterBase.tsx [hunk 2]
```diff
@@ -158,12 +159,18 @@ export interface IAttributeFilterBaseProps {
...
     /**
-     * Customize Customize attribute filter with a component dropdown button component
+     * Customize attribute filter button component
      */
     AttributeFilterButton?: React.ComponentType<IAttributeFilterButtonProps>;
 
+    /**
+     * Customize attribute filter dropdown actions buttons
+     */
+    AttributeFilterDropdownButtons?: React.ComponentType<IAttributeFilterDropdownButtonsProps>;
```
[the remark this answers is not in the document]
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFilter@next/AttributeFilterBase.tsx:169]: Renamed to FilterDropdownButtons

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/AttributeFilterDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    subtitle: string;
+
+    selectedFilterOptions: IListItem[];
+
+    onDropdownOpenStateChanged: (isOpen: boolean) => void;
+    onApplyButtonClicked: () => void;
+
+    isAllFiltered: boolean; //TODO move to DropDown props
+    hasNoData: boolean; //TODO move to DropDown props
+    applyDisabled: boolean; //TODO move to DropDown props
```
[the remark this answers is not in the document]
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFilter@next/Components/AttributeFilterDropdown.tsx:35]: renamed

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/AttributeFilterDropdownBody.tsx (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFilter@next/Components/AttributeFilterDropdownBody.tsx:10]: renamed

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8579/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10431/ : FAILURE in 13m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8581/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10434/ : SUCCESS in 12m 24s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2687

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2466/ : SUCCESS in 1m 04s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2762/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1588/ : SUCCESS in 17m 29s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10440/ : FAILURE in 9m 57s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8592/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10445/ : SUCCESS in 12m 47s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2687

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2467/ : SUCCESS in 58s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2763/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1589/ : SUCCESS in 28m 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10447/ : SUCCESS in 13m 36s

**Pull request merged**